src/compiler/crystal/macros/methods.cr

```diff
@@ -2394,6 +2394,16 @@ private def interpret_array_or_tuple_method(object, klass, method, args, named_a
       end
       klass.new(object.elements + other_elements)
     end
+  when "-"
+    interpret_check_args(node: object) do |arg|
+      case arg
+      when Crystal::ArrayLiteral
+        other_elements = arg.elements
+      else
+        arg.raise "argument to `#{klass}#-` must be array, not #{arg.class_desc}:\n\n#{arg}"
+      end
+      klass.new(object.elements - other_elements)
+    end
```

Comment:
Defining the method here means that this will also works with tuples, but I'm not sure we'd like that.

Furthermore, it will work for tuples, but only if you pass an array literal as the right-hand side operand.

I think this method definition should be moved to `ArrayLiteral`, or something like that.

Reply:
For consistency (but unfortunately introducing a breaking change), we could [remove the current behaviour to add a `TupleLiteral` to an `ArrayLiteral`, and vice-versa](https://carc.in/#/r/dxas).

This PR would make sense with respect to the current `+` behaviour between `ArrayLiteral` and `TupleLiteral` if it supported `TupleLiteral` as the RHS of the expression.

Reviving this PR because I reached for `ArrayLiteral#-` recently.